File: generated-code/cpp/codecraft/TcpReadWrite/Color.cpp
#include "Color.hpp"

Color::Color(float r, float g, float b, float a) : r(r), g(g), b(b), a(a) { }

// Read Color from input stream
Color Color::readFrom(InputStream& stream) {
    float r = stream.readFloat();
    float g = stream.readFloat();
    float b = stream.readFloat();
    float a = stream.readFloat();
    return Color(r, g, b, a);
}

// Write Color to output stream
void Color::writeTo(OutputStream& stream) const {
    stream.write(r);
    stream.write(g);
    stream.write(b);
    stream.write(a);
}

// Get string representation of Color
std::string Color::toString() const {
    std::stringstream ss;
    ss << "Color { ";
    ss << "r: ";
    ss << r;
    ss << ", ";
    ss << "g: ";
    ss << g;
    ss << ", ";
    ss << "b: ";
    ss << b;
    ss << ", ";
    ss << "a: ";
    ss << a;
    ss << " }";
    return ss.str();
}

File: generated-code/javascript/codecraft/TcpReadWrite/color.js
/**
 * RGBA Color
 */
class Color {
    /**
     * Red component
     */
    r;
    /**
     * Green component
     */
    g;
    /**
     * Blue component
     */
    b;
    /**
     * Alpha (opacity) component
     */
    a;

    constructor(r, g, b, a) {
        this.r = r;
        this.g = g;
        this.b = b;
        this.a = a;
    }

    /**
     * Read Color from input stream
     */
    static async readFrom(stream) {
        let r;
        r = await stream.readFloat();
        let g;
        g = await stream.readFloat();
        let b;
        b = await stream.readFloat();
        let a;
        a = await stream.readFloat();
        return new Color(r, g, b, a);
    }

    /**
     * Write Color to output stream
     */
    async writeTo(stream) {
        let r = this.r;
        await stream.writeFloat(r);
        let g = this.g;
        await stream.writeFloat(g);
        let b = this.b;
        await stream.writeFloat(b);
        let a = this.a;
        await stream.writeFloat(a);
    }
}
module.exports = Color